third_party/move/move-compiler-v2/src/function_checker.rs

```diff
@@ -253,6 +253,33 @@ pub fn check_access_and_use(env: &mut GlobalEnv, before_inlining: bool) {
                                 Visibility::Friend => {
                                     if callee_func.module_env.has_friend(&caller_module_id) {
                                         true
+                                    } else if callee_func.has_package_visibility() {
+                                        if callee_func.module_env.self_address()
+                                            == caller_func.module_env.self_address()
+                                        {
+                                            if callee_func
+                                                .module_env
+                                                .in_same_package(&caller_func.module_env)
+                                            {
+                                                // we should've inferred the friend declaration
+                                                panic!(
+                                                    "ICE: {} should have friend {}",
+                                                    callee_func.module_env.get_full_name_str(),
+                                                    caller_func.module_env.get_full_name_str()
+                                                );
+                                            } else {
+                                                diff_package_error(
```

Comment:
diff_package_error is not a very telling function (compared to not_a_friend_error), can we give this and diff_addr_error a better name?

Reply:
done